### PR TITLE
Add --target option

### PR DESCRIPTION
**Pull request by taiki-e**
Add support for a single `--target` option. (I did not implement multiple targets support because it seemed complicated.)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by taiki-e** [comment]
Hmm, this seems wrong for proc-macro.

**Comment by Velfi** [comment]
If you have any questions, let us know. Once you take this out of Draft status, we'll take a look.

**Comment by taiki-e** [comment]
> Hmm, this seems wrong for proc-macro.

Actually, proc-macro cannot define any public API other than `#[proc_macro]`/`#[proc_macro_attribute]`/`#[proc_macro_derive]`, so there is no need to use this tool for proc-macro.

In addition, the latest main rejects builds on on non-lib targets.
https://github.com/awslabs/cargo-check-external-types/blob/790a955d50f149290011e3dba1568a357af6feba/src/main.rs#L251-L256

File: src/main.rs [hunk 1]
```diff
@@ -387,6 +397,7 @@ mod arg_parse_tests {
                 no_default_features: false,
                 features: None,
                 manifest_path: None,
+                target: None,
```
Comment (jdisanti) [inline src/main.rs:400]: Can you add a test that exercises this new option?

Comment (taiki-e) [reply src/main.rs:423]: Added a test for this option.

**Review by jdisanti — APPROVED** [review]
Thank you! ❤️

**Pull request merged**